### PR TITLE
#99 render stacked bars

### PR DESCRIPTION
**Pull request by b4rdha**
#99  Added functionality of applying color by, so rendering stacked bars which a bar is composed of.

### CODE REVIEW

**Comment by MajaLinder** [comment]
Nice work @b4rdha! I noticed that the colors are reversed compared to the bar chart
image

**Comment by b4rdha** [comment]
> Nice work @b4rdha! I noticed that the colors are reversed compared to the bar chart image

Ah good catch, I missed it when I pushed, I fixed it now thanks!

**Review by MajaLinder — APPROVED** [review]

File: src/stackedBars.ts [hunk 1]
```diff
@@ -1,39 +1,53 @@
...
+    const paretoCategoryValues: string[] = moduleCategories(pareto);
+
+    const svg: any = document.querySelector("#svg");
+    const svgBoundingClientRect: any = svg.getBoundingClientRect();
+    const ticks = moduleTicks(svgBoundingClientRect.height, settings.style.label.size);
+    const categoryAxis = moduleCategoryAxis(paretoCategoryValues, 0, svgBoundingClientRect.width);
+    const valueAxis = moduleValueAxis(pareto.maxValue, svgBoundingClientRect.height, ticks);
+
+    let colorKeys = pareto.colorIndices.map((x) => (x ?? 0).toString());
+    console.log(colorKeys);
```
Comment (AstridBer) [inline src/stackedBars.ts:21]: Can the console.log be removed?

**Review by AstridBer — COMMENTED** [review]
Great work! So nice to finally have this! 😄 

I tried to run the code and everything seems to work well. But I noticed that if the ColorBy is "none" then the hardcoded value is used instead of the value provided by the api. In the picture below the color of the bars should be orange instead of blue.
colorByDefaultColor

**Comment by AstridBer** [comment]
Another thing that I noticed is that there is some strange behaviour when you mark a bar. I know this will be adressed in another issue but I think it is good to mention.

It seems like the pareto marks based on the color. In this picture you can compare how the built in visulaization reacts compared to the pareto.
MarksByColor

If I however mark the "first color" nothing is marked in the pareto. 
Doesn'tMarkFirstValue

Does anyone have any idea why this happens?

**Review by MajaLinder — CHANGES_REQUESTED** [review]
Before merging, I think we should go through Astrid's findings.

**Comment by MajaLinder** [comment]
I found another issue. When adding two color by, the bars are not rendering in the correct order, so maybe something is wrong with how everything is rendered?

image

**Review by MajaLinder — CHANGES_REQUESTED** [review]
After the merge the padding on the right side of the chart disappeared

image

**Review by MajaLinder — APPROVED** [review]

**Pull request merged**